impeller/renderer/frame_captor.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/macros.h"
+
+namespace impeller {
+
+//------------------------------------------------------------------------------
+/// @brief      Frame capture configuration.
+///
+struct FrameCaptorConfiguration {
+  /// This param is for metal backend.
+  /// When this value is true, a gpu trace file will be saved in devices when
+  /// capture finishes. Otherwise, the Xcode will automatically open and show
+  /// trace result.
+  ///
+  bool mtlSaveGPUTraceDocument = false;
+};
+
+//------------------------------------------------------------------------------
+/// @brief      A class used for frame capture during rendering. Backend like
+///             Metal, has a ability to capture frame programmatically.
+///             The frame capture should only be used in development for frame
+///             offline analysis.
+///
+class FrameCaptor {
```

Comment:
Nit: Can we rename this to something like `GPUTracer`? Since we will be using this to trace more than just frames (like compute workloads).

Reply:
Done^_^